### DISCUSSION TITLE
Problem in accumulating normal equations over a long time span

### DISCUSSION THREAD

**Discussion by ParisaShafiei** [groops-devs/groops#169] [Q&A]
Dear developers, 

I have one question regarding the normals accumulation. When it comes to accumulating normal equations over a long time span, how we should do it?

I have the normal equations written on a daily basis and now I wish to accumulate them over a long time span, e.g. 1 month. To do so, I used "NormalsAccumulate", but I did not succeed to get the accumulated normals. 

Actually, when I want to accumulate them over one month, I got the error shown in the following screenshot.I also tried another option, but did not work either: 
First accumulate the splitted normal in each day and write a daily normal equation file. 
Second accumulate the daily normal equations from step one over the desired time span.

I would be pleased if you could take a look and tell me how I can do it properly. 

Thank you in advance for your time and help.

With kind regards 
Parisa

![image](https://user-images.githubusercontent.com/99748448/224746081-4ae0de31-bbbf-43bc-9bfd-6556a2caf953.png)

![image](https://user-images.githubusercontent.com/99748448/224746405-216619f0-cf8e-4530-84cf-a2497422baea.png)

**Comment by tmayerguerr** [comment]
Dear Parisa,

NormalsAccumulate is quite simple. It assumes that all normal equations contain the same parameters in the same order. 
You should think about which parameters change every day (e.g. earth rotation or orbit parameters). You should eliminate these before. Also, not all stations are observed on all days.

The general steps are:
1. Create a file with parameter names which should be in the accumulated normals:
   [ParameterNamesCreate](https://groops-devs.github.io/groops/html/ParameterNamesCreate.html) -> selection (e.g. wildCard->type=position*: only stations positions) -> withoutDuplicates -> file (parameter names from daily normals with daily loop)
2. daily loop: [NormalsEliminate](https://groops-devs.github.io/groops/html/NormalsEliminate.html) -> remainingParameters->names->file (from step 1)
3. [NormalsAccumulate](https://groops-devs.github.io/groops/html/NormalsAccumulate.html) (input with daily loop)

regards
Torsten

**Reply by ParisaShafiei** [reply]
Dear Torsten, 

I have understood what was the problem. I had  parameters differing from one day to another. 

One question from my side:
I did the first step to keep only the receiver antenna parameters and write them in a parameter name file (please look at the screenshot). However, the problem is this new parameters name file still contains all the parameters and when I used it in "NormalsEliminate" it says that no parameters are eliminated. 

![image](https://user-images.githubusercontent.com/99748448/224992058-a8c40171-5830-4640-b0af-5b120460ba16.png)

![image](https://user-images.githubusercontent.com/99748448/224994126-5e7aa5b6-5bbd-42d8-8ec6-d7b8106146bc.png)


I tried another option to just use parametrizationGnssAntenna to write down the antenna related parameters, it also failed with an error shown in the screenshot. 

![image](https://user-images.githubusercontent.com/99748448/224993424-57f94084-dc70-4910-ad46-06e223d21c21.png)

![image](https://user-images.githubusercontent.com/99748448/224995120-177f4ba7-c3fc-45b2-9ba7-f5c4fb1d9e07.png)

Thank you for your time and support. 

With kind regards, 
Parisa

**Reply by tmayerguerr** [reply]
Could you please send me one file output/*paramerterNames .txt of a single day?
Torsten

**Reply by ParisaShafiei** [reply]
Hi Torsten, 

Please find the attached normals_2020-01-01.parameterNames.txt. 
I should just keep the antennaCenterVariations.sphericalHarmonics* in the normal equations and eliminate all the other parameters like signal biases, TEC and so on, am I right? 

[normals_2020-01-01.parameterNames.txt](https://github.com/groops-devs/groops/files/10979246/normals_2020-01-01.parameterNames.txt)

Thanks for checking out the attached file. 

With kind regards, 
Parisa

**Comment by ParisaShafiei** [comment]
Dear Torsten, 

As an update, I tried another option and this is what I got:

1. [ParameterNamesCreate](https://groops-devs.github.io/groops/html/ParameterNamesCreate.html) > parametrizationGnssAntenna > sphericalHarmonics with maxdegree = 30
2. [NormalsEliminate](https://groops-devs.github.io/groops/html/NormalsEliminate.html) > remaining parameters (only antenna parametrization) with outBlockSize = 5000, I managed to get the eliminated normals and it did not give the error that it cannot compute the decomposition

But now, my problem is after following the all the steps, the ACV patterns I got for GRACE FO-C with using one month of data is totally different from the original ACVs. Is there somthing else that I need to consider in my processing to get an ACVs at least close to pattern of the original one?

![ACV_pattern_test](https://user-images.githubusercontent.com/99748448/225907582-533d28fe-ac14-46be-a25e-2609e2372f70.png)

Thank you in advance for the time you spend on answering our questions. It is really appreciated. 

With kind regards, 
Parisa
